hip-0000-exchange-rate-system-contract.md

```diff
@@ -0,0 +1,128 @@
+
+## Motivation
+
+Suppose a contract aspires to be "self-funding"---that is, it expects users to send enough msg.value with function calls 
+to cover the contract's Hedera fees.
+
+The problem is that, even though Hedera fees are fixed and predictable, they are denominated in tinycents (USD); but 
+`msg.value` is denominated in tinybars.  So the contract needs to convert from tinycents to tinybars, at the 
+_same exchange rate_ the network will use when charging fees.
+
+## Rationale
+
+There are not many degrees of freedom in the design of a component that performs one multiplication and one division; 
+our main goal was to mirror the prior art that began in [HIP-206](https://hips.hedera.com/hip/hip-206). Thus the function 
+selectors that determine whether the contract converts from tinycents to tinybars, or vice-versa, are computed as the
+function selectors of the Solidity interface in the below specification.
+ 
+## User stories
+
+- As the author of a "self-funding" contract, I want to require users to include enough `msg.value` to cover my contract's Hedera fees.
+- As the deployer of a contract that holds large hbar reserves, I want a `view` function that returns the approximate USD value held by my contract.
+  
+## Specification
+
+The contract is to reside at address `0x168` and implement the below interface,
+```
+interface IExchangeRate {
+    // Given a value in tinycents (1e-8 US cents or 1e-10 USD), returns the 
+    // equivalent value in tinybars (1e-8 HBAR) at the current exchange rate 
+    // stored in system file 0.0.112. 
+    // 
+    // This rate is a weighted median of the the recent" HBAR-USD exchange 
+    // rate on major exchanges, but should _not_ be treated as a live price 
+    // oracle! It is important primarily because the network will use it to 
+    // compute the tinybar fees for the active transaction. 
+    // 
+    // So a "self-funding" contract can use this rate to compute how much 
+    // tinybar its users must send to cover the Hedera fees for the transaction.
+    function tinycentsToTinybars(uint256 tinycents) external returns (uint256);
+
+    // Given a value in tinybars (1e-8 HBAR), returns the equivalent value in 
+    // tinycents (1e-8 US cents or 1e-10 USD) at the current exchange rate 
+    // stored in system file 0.0.112. 
+    // 
+    // This rate tracks the the HBAR-USD rate on public exchanges, but 
+    // should _not_ be treated as a live price oracle! This conversion is
+    // less likely to be needed than the above conversion from tinycent to
+    // tinybars, but we include it for completeness.
+    function tinybarsToTinycents(uint256 tinybars) external returns (uint256);
+}
+```
+
+## Backwards Compatibility
+
+There are no known sources of backward incompatibility.
+
+## Security Implications
+
+There are no obvious security implications. Even if the exchange rate in file `0.0.112` diverges from the USD-HBAR exchange 
```

Comment:
"There are no obvious security implications." famous last words :)

I think we need to enumerate the risk of an exchange depending on this precompile as an exchange oracle.  Even though we call that out as a bad idea we need to explain potential impacts and mitigations. Exposing this info to smart contracts is a new risk surface.

Reply:
Highlighted this risk.